crates/router/src/core/blocklist/utils.rs

```diff
@@ -357,3 +361,169 @@ async fn delete_card_bin_blocklist_entry(
+    let card_bin_fingerprint = payment_method_data
+        .as_ref()
+        .and_then(|pm_data| match pm_data {
+            api_models::payments::PaymentMethodData::Card(card) => {
+                crypto::HmacSha512::sign_message(
+                    &crypto::HmacSha512,
+                    merchant_fingerprint_secret.as_bytes(),
+                    card.card_number.clone().get_card_isin().as_bytes(),
+                )
+                .attach_printable("error in card bin hash creation")
+                .map_or_else(
+                    |err| {
+                        logger::error!(error=?err);
+                        None
+                    },
+                    Some,
+                )
+            }
+            _ => None,
+        })
+        .map(hex::encode);
+
+    // Hashed Extended Cardbin to check whether or not this payment should be blocked.
+    let extended_card_bin_fingerprint = payment_method_data
+        .as_ref()
+        .and_then(|pm_data| match pm_data {
+            api_models::payments::PaymentMethodData::Card(card) => {
+                crypto::HmacSha512::sign_message(
+                    &crypto::HmacSha512,
+                    merchant_fingerprint_secret.as_bytes(),
+                    card.card_number.clone().get_extended_card_bin().as_bytes(),
+                )
+                .attach_printable("error in extended card bin hash creation")
+                .map_or_else(
+                    |err| {
+                        logger::error!(error=?err);
+                        None
+                    },
+                    Some,
+                )
+            }
+            _ => None,
+        })
+        .map(hex::encode);
+
+    let mut fingerprint_id = None;
+
+    //validating the payment method.
+    let mut is_pm_blocklisted = false;
+
+    let mut blocklist_futures = Vec::new();
+    if let Some(card_number_fingerprint) = card_number_fingerprint.as_ref() {
+        blocklist_futures.push(db.find_blocklist_lookup_entry_by_merchant_id_fingerprint(
+            &merchant_id,
+            card_number_fingerprint,
+        ));
+    }
+
+    if let Some(card_bin_fingerprint) = card_bin_fingerprint.as_ref() {
+        blocklist_futures.push(db.find_blocklist_lookup_entry_by_merchant_id_fingerprint(
+            &merchant_id,
+            card_bin_fingerprint,
+        ));
+    }
+
+    if let Some(extended_card_bin_fingerprint) = extended_card_bin_fingerprint.as_ref() {
+        blocklist_futures.push(db.find_blocklist_lookup_entry_by_merchant_id_fingerprint(
+            &merchant_id,
+            extended_card_bin_fingerprint,
+        ));
+    }
+
+    let blocklist_lookups = futures::future::join_all(blocklist_futures).await;
+
+    if blocklist_lookups.iter().any(|x| x.is_ok()) {
+        intent_status = storage_enums::IntentStatus::Failed;
+        attempt_status = storage_enums::AttemptStatus::Failure;
+        is_pm_blocklisted = true;
+    }
```

Comment:
Update this to log any DB errors that might have been returned.
```suggestion
    if blocklist_lookups.iter().any(|x| x.is_ok()) {
        intent_status = storage_enums::IntentStatus::Failed;
        attempt_status = storage_enums::AttemptStatus::Failure;
        is_pm_blocklisted = true;
    }
```